### PR TITLE
Candidate submission

### PR DESCRIPTION
**Pull request by nateallen-wk**
From the candidate:

> Here is the code. I had a lot of fun with it and kinda just let it come naturally.

### CODE REVIEW

File: css/styles.css [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    color: #bb1324;
+}
+
+.has-error {
+    background: #f8d7da !important;
+    border: 1px solid #d4868d !important;
+}
+
+#alert-container {
+
```
Comment (laurameckley-wk) [inline css/styles.css]: undefined css style

File: css/styles.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    font-weight: 700;
+}
+
+.required {
+    color: #bb1324;
+}
+
+.has-error {
+    background: #f8d7da !important;
+    border: 1px solid #d4868d !important;
```
Comment (laurameckley-wk) [inline css/styles.css]: using !important is not ideal

File: css/styles.css [hunk 3]
```diff
@@ -0,0 +1,64 @@
+html {
```
Comment (laurameckley-wk) [inline css/styles.css]: reasoning behind creating a new css file?

File: pages/main.html [hunk 4]
```diff
@@ -0,0 +1,83 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+
+  <meta charset="utf-8">
+  <title>JIRA Quick Search</title>
+  <meta name="description" content="">
+  <meta name="author" content="">
```
Comment (laurameckley-wk) [inline pages/main.html]: blank metas

File: README.md [hunk 5]
```diff
@@ -1,29 +1,54 @@
...
+* See if there is a json based api for activity feed and put a bullet in XML (yuck)
+* Enforce the max on the date field. Error handling captures it though ;)
+* Oh god, use constants for urls lol
 
-## Tips
-* [Installing a local chrome extension](https://developer.chrome.com/extensions/getstarted#unpacked)
-* The first field in the extension should default to the word "Sunshine", if it does not, it means there is a bug that needs to be fixed first.
\ No newline at end of file
+## Changes:
+* Fixed - Removed async call in a non-async function
```
Comment (shanesawyer-wf) [inline README.md]: Was this the missing async in checkProjectExists?

File: README.md [hunk 6]
```diff
@@ -1,29 +1,54 @@
...
+  * Tried to make both bindings closer to common, but xml vs json.. also changed domify
+* Fixed - Added some "common" datarow rendering
+* Fixed - Window does not flex out on results, should state somewhat static minus scrollbar
+* General - Organized files in solution, more maintainable
+* General - Moved events and page setup stuff to top of files
+* General - Consistent naming on functions
+* General - Commented all functions, intellisense weeee
+* General - Changed a lot to string interpolation
+* General - Moved http into common function
+* General - Moved to promises, I like them better
```
Comment (shanesawyer-wf) [inline README.md]: Why do you like promises vs async/await?

File: pages/main.html [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+    <!-- TABS -->
+    <div class="tab">
+      <button class="tablinks" id="tab-status">Ticket Status</button>
+      <button class="tablinks" id="tab-activity">Activity Query</button>
+    </div>
+    
+    <!-- TAB -->
+    <div id="tab-status" class="tabcontent open">
+        <label>Project name <span class="required">*</span></label>
```
Comment (laurameckley-wk) [inline pages/main.html]: What is the reasoning behind requiring the project name? What if the user wants to query all projects that have in progress tickets for example?

File: scripts/main.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+// Main Functions
+// -------------------------------------------------------------
+
+document.addEventListener('DOMContentLoaded', setupPage);
+
+/**
+ * Sets up the page and event handlers
+ */
+function setupPage() {
+    // LOAD DEFAULT OPTIONS
```
Comment (laurameckley-wk) [inline scripts/main.js]: Could give a few examples of things in default settings here

File: js/fields.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Validates a field and makes it run
+ * @param {elementName} elementName - The element name to make red
+ */
+function validateField(elementName) {
```
Comment (laurameckley-wk) [inline js/fields.js]: Reasoning behind this being in its own file?

**Comment by markmadej-wk** [comment]
If you restart your browser, open the plugin, and search for "SUN" as shown, an error occurs.
screen shot 2018-02-09 at 7 20 48 am

File: README.md [hunk 10]
```diff
@@ -1,29 +1,54 @@
...
+A chrome extension that queries a public JIRA API (https://jira.secondlife.com). 
 
 ## Files:
-* main.js - the :meat_on_bone: of the code that performs querys and displays results
-* main.html - The UI for the request results
-* manifest.json - contains metadata for the project and defines access permissions
-* options.js - contains logic for saving/retrieving user values
-* options.html - display for the user values
-* json_results/results.json - a sample of the json values returned from a JIRA ticket(s) query
+* **scripts/main.js** - Contians the main logic for the extension
```
Comment (markmadej-wk) [inline README.md]: Typo - `Contians`

File: README.md [hunk 11]
```diff
@@ -1,29 +1,54 @@
...
+* **scripts/options.js** - Contains the logic for saving and defaulting prefs
+* **pages/main.html** - The UI for the main page.
+* **pages/options.html** - display for the user values
+* **js/alerts.js** - Common alert functionality
+* **js/fields.js** - Common field functionality
+* **js/tabs.js** - Common tabs functionality
+* **js/http.js** - Common http/get functionality
+* **js/functions.js** - Common used functions
+* **manifest.json** - contains metadata for the project and defines access permissions
```
Comment (markmadej-wk) [inline README.md]: #nit Inconsistent uppercase/lowercase on a few of the file descriptions' first word

File: js/http.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+        req.onload = function() {
+            var response = req.responseXML ? req.responseXML : req.response;
+            if (response && response.errorMessages && response.errorMessages.length > 0) {
+                reject(response.errorMessages[0]);
+                return;
+            }
+            resolve(response);
+        };
+
+        req.onerror = function(error) {
```
Comment (chrishague-wf) [inline js/http.js]: NOTE TO SELF: Ask candidate why we have this error handler here and an error event listener blow (line 30)

File: js/tabs.js [hunk 13]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Selects and renders a tab
+ * @param {tabElement} tabElement - The element that triggered the change
+ */
+function openTab(tabElement) {
+    var tabName = tabElement.srcElement.id;
+    var tabs = document.getElementsByClassName("tabcontent");
+
+    for (i = 0; i < tabs.length; i++) {
```
Comment (chrishague-wf) [inline js/tabs.js]: paranoid null check on tabs perhaps?

**Pull request closed without merging**